src/main/java/net/smartcosmos/dao/things/impl/ThingPersistenceService.java

```diff
@@ -158,12 +158,10 @@ public ThingPersistenceService(
 
         UUID id = UuidUtil.getUuidFromUrn(urn);
 
-        Optional<ThingEntity> entity;
+        Optional<ThingEntity> entity = Optional.empty();
         if (StringUtils.isNotBlank(tenantUrn)) {
```

Comment:
Why is this not `IllegalArgumentException`? An invalid or empty urn should be IMO.  Also the logic here is a little off, it should look like this
```
Assert.hasLength(tenantUrn);

UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
Optional<ThingEntity> entity = repository.findByIdAndTenantIdAndType(UuidUtil.getUuidFromUrn(urn), tenantId, type);
if (entity.isPresent()) {
       return Optional.ofNullable(conversionService.convert(entity.get(), ThingResponse.
}
        
return Optional.empty();
```